syntaxes/mermaid.tmLanguage-markdown.yaml

```diff
@@ -3,13 +3,19 @@ fileTypes: []
 injectionSelector: L:text.html.markdown
 patterns:
   - include: '#mermaid-code-block'
+  - include: '#mermaid-code-block-with-attributes'
   - include: '#mermaid-ado-code-block'
 repository:
   mermaid-code-block:
     begin: (?<=[`~])mermaid(\s+[^`~]*)?$
     end: (^|\G)(?=\s*[`~]{3,}\s*$)
     patterns:
       - include: '#mermaid'
+  mermaid-code-block-with-attributes:
+    begin: (?<=[`~])\{ *\.mermaid(\s+[^`~]*)?$
```

Comment:
```
begin: (?<=[`~])\{\s*\.mermaid(\s+[^`~]*)?$
```

^ suggest changing your `space` to \s for readability

Reply:
Going to answer my own question @bpruitt-goddard. It looks like the `.` in `{.mermaid caption="Title" #fig:graph}`  is required.  While researching, I found a similar issue in the pandox repo itself. https://github.com/Siphalor/pandoc/commit/8757b9e70f258bc1a2038355ab4f30a5efcdbd6a#diff-9affa84862db56128c85f089a8695ffa091f426415a97d2cb8ee8bc6ba5778bdR3. It looks like when you write it as:

```
```mermaid {caption="Title" #fig:graph}

<-- SNIP -->
```

both syntax highlighting and rendering work as expected.

That being said, I still think its a net good for little effort to support the syntax highlighting.

Reply:
@mattorp Are you able to fix the merge, or would you like me to carry on in my own PR?